dbms/src/Storages/DeltaMerge/RowKeyRange.h

```diff
@@ -84,6 +84,42 @@ struct RowKeyValue
         {
             size_t cursor = 0;
             int_value = DB::DecodeInt64(cursor, *value);
+            if (unlikely(value->size() != sizeof(Int64)))
+            {
+                // For int type handle, the standard key enconding format should be t{table_id}_r{handle_value}.
+                // But TiKV may generate region range keys which are not strictly following the standard format.
+                // More concretely, the key may be t{table_id}_r{handle_value} + some other bytes.
+                // We need to adapt the key to the standard format.
+                // For example, the key may be t100_r1000 + 0x00, we need to adapt it to t100_r1001.
+                // This is ok, because
+                //  1) if the key is the start range, then [t100_r1000 + 0x00, xxx) has the same semantics with [t100_r1001, xxx)
+                //  2) if the key is the end range, then [xxx, t100_r1000 + 0x00) also has the same semantics with [xxx, t100_r1001)
+                //
+                // Note if the `int_value` is Int64::max_value,
+                // it is a value generated by tiflash itself to means +inf()(which is RowKeyValue::INT_HANDLE_MAX_KEY).
+                // So we can just ignore it.
+                if (value->size() == sizeof(UInt64) + 1)
+                {
+                    if (int_value < std::numeric_limits<Int64>::max() && value->back() == 0x00)
```

Comment:
Why don't we need an  "else" here ?

Reply:
Refactor the structure to make it more clear.